samples/web/content/apprtc/js/appwindow.js

```diff
@@ -8,63 +8,64 @@
 
 /* More information about these options at jshint.com/docs/options */
 // Variables defined in and used from main.js.
-/* globals randomString, AppController, UI_CONSTANTS */
+/* globals randomString, AppController, sendAsyncUrlRequest, parseJSON */
 /* exported params */
 'use strict';
 
-// Provide default params set to the values returned by apprtc.appspot.com.
-var params = {
-  errorMessages: [],
-  isLoopback: false,
-  mediaConstraints: {
-    'audio': true,
-    'video': {
-      'optional': [{
-        'minWidth': '1280'
-      }, {
-        'minHeight': '720'
-      }],
-      'mandatory': {}
-    }
-  },
-  offerConstraints: {
-    'optional': [],
-    'mandatory': {}
-  },
-  peerConnectionConfig: {
-    'iceServers': []
-  },
-  peerConnectionConstraints: {
-    'optional': [{
-      'googImprovedWifiBwe': true
-    }]
-  },
-  turnRequestUrl: 'https://computeengineondemand.appspot.com/turn?username=073557600&key=4080218913',
-  turnTransports: '',
-  audioSendBitrate: '',
-  audioSendCodec: '',
-  audioRecvBitrate: '',
-  audioRecvCodec: '',
-  isStereoscopic: '',
-  opusMaxPbr: '',
-  opusFec: '',
-  opusStereo: '',
-  videoSendBitrate: '',
-  videoSendInitialBitrate: '',
-  videoSendCodec: '',
-  videoRecvBitrate: '',
-  videoRecvCodec: '',
-  wssUrl: 'wss://apprtc-ws.webrtc.org:443/ws',
-  wssPostUrl: 'https://apprtc-ws.webrtc.org:443'
-};
-
 // Generate random room id and connect.
-params.roomId = randomString(9);
-params.roomLink =  'https://apprtc.appspot.com/room/' + params.roomId;
-params.roomServer = 'https://apprtc.appspot.com';
 
-var joinRoomLink = document.querySelector(UI_CONSTANTS.roomLinkHref);
-joinRoomLink.href = params.roomLink;
-joinRoomLink.text = params.roomLink;
+var roomServer = 'https://apprtc.appspot.com';
+var loadingParams = {
+  errorMessages: [],
+  roomId: randomString(9),
+  roomServer: roomServer,
+  paramsFunction: function() {
+    return new Promise(function(resolve, reject) {
+      trace('Initializing; retrieving params from: ' + roomServer + '/params');
+      sendAsyncUrlRequest('GET', roomServer + '/params').then(function(result) {
+        var serverParams = parseJSON(result);
+        var newParams = {};
+        if (!serverParams)
+        {
+          resolve(newParams);
```

Comment:
Instead of resolve? Wouldn't the promise we are returning never settle as a result?